cmd.js

```diff
@@ -98,12 +98,42 @@ function readFile(path) {
   });
 }
 
+function fixupDefines(flags){
+ /**
+   * compile() expects 'defines' to be object, but minimist may only return array or string
+   * so, we need to convert to proper object.
+   * The supported format will be similar to the following,
+   * https://github.com/google/closure-compiler/wiki/Annotating-JavaScript-for-the-Closure-Compiler#define-type-description
+   * but instead of using --define 'name=value' we should support --defines 'name=value'
+   */
+
+  if (Array.isArray(flags.defines) || typeof flags.defines === "string"){
```

Comment:
flags.defines should always be either a string or an array right? So should this `if` have an `else { throw new Error('Invalid value for defines'); }`?

Reply:
it is an optional field. So, it can also be undefined or null. 
This check basically also checks if "flags.defines has value" since those values can only be string or array.
correct?

Reply:
Oh of course, I forgot about undefined/null. Let's add a test for those as well (but we should still fail on anything that isn't undefined, null, an array, or a string).

Reply:
done